### PR TITLE
Revised build process to use bundled publishing

### PR DESCRIPTION
**Pull request by davesmith00000**
Should work, I haven't tried an actual bundle release yet. All builds should be faster and easier to maintain as a side effect.

The next trick will be to include a standard version management mechanism. After that moving to travis should be pretty easy.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -432,14 +422,19 @@ lazy val docs =
...
+      sourceDirectory in Paradox := sourceDirectory.value / "main" / "paradox",
+      skip in publish := true
     )
 
 lazy val scalaPactProject =
   (project in file("."))
     .settings(commonSettings: _*)
-    .aggregate(shared, core, pluginShared, plugin, pluginNoDeps, framework, standalone, testShared)
+    .settings(
+      skip in publish := true
```
Comment (dantb) [inline build.sbt]: What do we need this OOC?

File: scripts/release.sh [hunk 2]
```diff
@@ -25,11 +25,19 @@ else
 fi
 
 echo ""
-echo "Starting Stage 1 Deploy"
+echo "Preparing Scala-Pact for publishing"
 
-bash scripts/release-stage1-libs.sh
+sbt prepareScalaPactPublish
```
Comment (dantb) [inline scripts/release.sh]: What exactly is happening here? Will this `publishSigned` every sub-project to the remote staging repo? Then when we do the `sonatypeBundleRelease` pull them down locally, sign them, then release in one upload? Or can we do the whole thing locally until the signed artefact upload

File: build.sbt [hunk 3]
```diff
@@ -1,67 +1,54 @@
...
-  "-language:existentials", // Existential types (besides wildcard types) can be written and inferred
-  "-language:experimental.macros", // Allow macro definition (besides implementation and application)
-  "-language:higherKinds", // Allow higher-kinded types
-  "-language:implicitConversions", // Allow definition of implicit functions called views
-  "-unchecked", // Enable additional warnings where generated code depends on assumptions.
-  "-Xcheckinit", // Wrap field accessors to throw an exception on uninitialized access.
-  "-Xfatal-warnings", // Fail the compilation if there are any warnings.
-  "-Xfuture", // Turn on future language features.
-  "-Xlint:adapted-args", // Warn if an argument list is modified to match the receiver.
+  "utf-8",                            // Specify character encoding used by source files.
```
Comment (dantb) [inline build.sbt]: The work of @nickgravgaard's tab stop tool by any chance? 😄

Comment (davesmith00000) [reply build.sbt]: Scalafmt, have they adopted elastic tabstops? :P

File: build.sbt (hunk 1 above)
Comment (davesmith00000) [reply build.sbt]: SBT by default will try and publish all the things, including the root project which clearly will not publish anything. When it tries to publish the root project with nothing in it, you get an error. There are many flavours of "don't publish this thing" but this method seems to be the only one that works for all modules *including* the root project.

File: scripts/release.sh (hunk 2 above)
Comment (davesmith00000) [reply scripts/release.sh]: No it's doing what you want it to do.
If you're artefacts are marked as SNAPSHOT then the publishSigned process uploads them to sonatype one by boring one as normal.
For non-SNAPSHOT artefacts, the signed artefacts are gathered in `<root>/target/sonatype-staging/` locally on your machine, and then bulk uploaded.

File: commands.sbt [hunk 4]
```diff
@@ -0,0 +1,21 @@
+addCommandAlias(
```
Comment (agustafson) [inline commands.sbt]: nice idea to break these out into an appropriately-named file 👍

**Pull request merged**